### PR TITLE
2300 Add support to decollate non-batch data

### LINKED ISSUE #2300: decollate needs to support non-batch data
**Is your feature request related to a problem? Please describe.**
Currently, if we apply `decollate` to a data, every item must have same batch size.
We need to support the case that some items may be `None` or scalar value, like `loss`.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #2300 .

### Description
This PR added support to decollate the scalar data in a list or dict.
Also removed the `Decollated` transform we aligned to only support `channel-first` data in all the transforms now.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/data/utils.py [hunk 1]
```diff
@@ -288,7 +289,11 @@ def list_data_collate(batch: Sequence):
         raise TypeError(re_str)
 
 
-def decollate_batch(data: Union[dict, list, torch.Tensor], batch_size: Optional[int] = None) -> List[dict]:
+def decollate_batch(
+    data: Union[dict, list, torch.Tensor],
+    batch_size: Optional[int] = None,
+    copy_non_batch: bool = True,
+) -> List[dict]:
```
Comment (yiheng-wang-nv) [inline monai/data/utils.py:296]: `) -> List[dict, list, torch.Tensor]:`

Comment (Nic-Ma) [reply monai/data/utils.py]: Sure, updated in the latest commit.
Thanks.

**Comment by wyli** [comment]
could you merge this into #2244 instead of `dev` directly? the separate PRs make it difficult to release potential hotfixes and the new major features are not organised. I think we've discussed avoiding such situations.

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Sure, I will close this PR, and already submitted PR to #2244 : https://github.com/Nic-Ma/MONAI/pull/120

Thanks.

**Pull request closed without merging**